File: tests/cpp/arrays/test_min_rewards.cpp
#include "../../../problems/arrays/min_rewards/cpp/min_rewards.h"
#include "../jsontestbase.h"

#include <vector>

class TestMinRewards : public JsonTestBase {
public:
  TestMinRewards() {
    json_file_path = std::string(TEST_CASES_DIR) + "/arrays/min_rewards.json";
  }
};

TEST_F(TestMinRewards, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty or not loaded";

  for (const auto &test : test_cases) {
    auto scores = test["scores"].get<std::vector<int>>();
    auto expected = test["expected"].get<int>();

    auto result = minRewards(scores);

    EXPECT_EQ(result, expected) << "Failed for test case: " << test.dump(2);
  }
}

File: tests/python/arrays/min_rewards/test_min_rewards.py
from problems.arrays.min_rewards.python.min_rewards import minRewards
from tests.python.utility import load_test_cases


def test_min_rewards() -> None:
    test_cases = load_test_cases("arrays/min_rewards.json")

    for idx, test in enumerate(test_cases):
        scores: list[int] = test["scores"]
        expected: int = test["expected"]

        result: int = minRewards(scores)

        assert (
            result == expected
        ), f"Test case {idx} failed: expected {expected}, got {result}"
